

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Assembly summary and statistics &mdash; pieris-supplement 1.0 documentation</title>
  

  
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="pieris-supplement 1.0 documentation" href="index.html"/>
        <link rel="next" title="Meta-data analysis" href="meta-data.html"/>
        <link rel="prev" title="Transcriptome supplemental materials for Pieris brassicae" href="index.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> pieris-supplement
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">The assembly</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#sequence-preprocessing">Sequence preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-trinity">Running Trinity</a></li>
<li class="toctree-l2"><a class="reference internal" href="#transcriptome-summary">Transcriptome summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="#map-the-reads-using-blast">Map the reads using BLAST</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#swissprot-isoforms">SwissProt (isoforms)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#blast-agains-all-known-d-melanogaster-proteins">BLAST agains all known D. melanogaster proteins</a></li>
<li class="toctree-l3"><a class="reference internal" href="#blast-against-all-know-d-plexippus">BLAST against all know D. plexippus</a></li>
<li class="toctree-l3"><a class="reference internal" href="#summary">Summary</a></li>
<li class="toctree-l3"><a class="reference internal" href="#links">Links</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="meta-data.html">Meta-data analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="diff-expression.html">Differential expression analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="gene-sets/gene-sets.html">Functional gene sets</a></li>
<li class="toctree-l1"><a class="reference internal" href="lepidoptera.html">Lepidoptera transcriptomes</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="methods/database.html">Database</a></li>
<li class="toctree-l1"><a class="reference internal" href="references.html">Works cited</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">pieris-supplement</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Assembly summary and statistics</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/de-novo-assembly.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="assembly-summary-and-statistics">
<h1>Assembly summary and statistics<a class="headerlink" href="#assembly-summary-and-statistics" title="Permalink to this headline">¶</a></h1>
<p>We carried out the <em>de novo</em> transcriptome assembly with the software
pacakge <a class="reference external" href="http://trinityrnaseq.sourceforge.net">Trinity</a>
[Grabherr11].  The Trinity software suite consists of three main
pieces:</p>
<blockquote>
<div><ul class="simple">
<li><strong>Inchworm</strong> - assembles the RNA-seq data into the unique
sequences of transcripts, often generating full-length transcripts
for a dominant isoform, but then reports just the unique portions of
alternative ly spliced transcripts.</li>
<li><strong>Chrysalis</strong> - clusters the Inchworm contigs into clusters and
constructs complete de Bruijn graphs for each cluster. Each
cluster represents the full transcriptonal complexity for a given
gene (or sets of genes that share sequences in common). Chrysalis
then partitions the full read set among these disjoint graphs.</li>
<li><strong>Butterfly</strong> then processes the individual graphs in parallel,
tracing the paths that reads and pairs of reads take within the
graph, ultimately reporting full-length transcripts for alternatively
spliced isoforms, and teasing apart transcripts that corresponds to
paralogous genes.</li>
</ul>
</div></blockquote>
<div class="section" id="sequence-preprocessing">
<h2>Sequence preprocessing<a class="headerlink" href="#sequence-preprocessing" title="Permalink to this headline">¶</a></h2>
<p>A Python script was used to unzip, concatenate and trim the original reads via system calls (<a class="reference external" href="https://docs.python.org/2/library/subprocess.html">subprocess</a>).
We used <a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a> [Bolger14] to trim reads based on quality with default settings (LEADING:5 TRAILING:5 MINLEN:36).</p>
<blockquote>
<div><ul class="simple">
<li><a class="reference download internal" href="_downloads/preprocessReads.py" download=""><code class="xref download docutils literal"><span class="pre">preprocessReads.py</span></code></a></li>
</ul>
</div></blockquote>
<p>There are four output files.  Two are for the <em>paired</em> output where both reads passed, and two are for corresponding <em>unpaired</em> output where only one read passed (see manual).</p>
</div>
<div class="section" id="running-trinity">
<h2>Running Trinity<a class="headerlink" href="#running-trinity" title="Permalink to this headline">¶</a></h2>
<p>A Python script was used to generate the Trinity arguments.</p>
<blockquote>
<div><ul class="simple">
<li><a class="reference download internal" href="_downloads/runTrinity.py" download=""><code class="xref download docutils literal"><span class="pre">runTrinity.py</span></code></a></li>
</ul>
</div></blockquote>
<p>The generalized Trinity command is shown below.</p>
<blockquote>
<div><div class="highlight-bash"><div class="highlight"><pre><span></span>~$ <span class="nb">export</span> <span class="nv">TRINITY_HOME</span><span class="o">=</span><span class="s2">&quot;/usr/src/trinityrnaseq-2.0.4&quot;</span>
~$ <span class="nv">$TRINITY_HOME</span>/Trinity --seqType fq --output /path/to/out --trimmomatic --full_cleanup
   --SS_lib_type FR --max_memory 26G --CPU <span class="m">29</span> --normalize_reads
   --left left1.fastq,left2.fastq,left3.fastq
   --right right1.fastq,right2.fastq,right3.fastq <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">|</span> tee ./run-trinity-gg.log
</pre></div>
</div>
</div></blockquote>
<p>This produces the output file <cite>./&lt;TRINITY_OUT&gt;/Trinity.fasta</cite> which we can run some basic statistics on.</p>
</div>
<div class="section" id="transcriptome-summary">
<h2>Transcriptome summary<a class="headerlink" href="#transcriptome-summary" title="Permalink to this headline">¶</a></h2>
<p>Trinity groups transcripts into clusters that are <em>loosely referred to as a gene</em>.  The accession identifiers in the <cite>./trinity_out_dir/Trinity.fasta</cite> encode gene and isoform information.  Per the documentation (see links below) if we have the accession <cite>&gt;c0_g1_i1</cite> this refers to Trinity read cluster <cite>c0</cite>, gene <cite>g1</cite> and isoform <cite>i1</cite>.  The gene identifier in this case is <cite>c0_g1</cite>.</p>
<p>The basic stats of the assembly are:</p>
<blockquote>
<div><div class="highlight-bash"><div class="highlight"><pre><span></span>~$ <span class="nb">export</span> <span class="nv">TRINITY_HOME</span><span class="o">=</span><span class="s2">&quot;/usr/src/trinityrnaseq-2.0.4&quot;</span>
~$ <span class="nv">$TRINITY_HOME</span>/util/TrinityStats.pl ~/sequencing/pieris/dn-trinity/Trinity.fasta


<span class="c1">################################</span>
<span class="c1">## Counts of transcripts, etc.</span>
<span class="c1">################################</span>
Total trinity <span class="s1">&#39;genes&#39;</span>:65012
Total trinity transcripts:98416
Percent GC: <span class="m">38</span>.04

<span class="c1">########################################</span>
Stats based on ALL transcript contigs:
<span class="c1">########################################</span>

Contig N10: <span class="m">4650</span>
Contig N20: <span class="m">3779</span>
Contig N30: <span class="m">3184</span>
Contig N40: <span class="m">2645</span>
Contig N50: <span class="m">2155</span>

Median contig length: <span class="m">595</span>
Average contig: <span class="m">1151</span>.58
Total assembled bases: <span class="m">113334017</span>


<span class="c1">#####################################################</span>
<span class="c1">## Stats based on ONLY LONGEST ISOFORM per &#39;GENE&#39;:</span>
<span class="c1">#####################################################</span>

Contig N10: <span class="m">4325</span>
Contig N20: <span class="m">3444</span>
Contig N30: <span class="m">2704</span>
Contig N40: <span class="m">2056</span>
Contig N50: <span class="m">1496</span>

Median contig length: <span class="m">403</span>
Average contig: <span class="m">807</span>.01
Total assembled bases: <span class="m">52465576</span>
</pre></div>
</div>
</div></blockquote>
</div>
<div class="section" id="map-the-reads-using-blast">
<h2>Map the reads using BLAST<a class="headerlink" href="#map-the-reads-using-blast" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>First, we BLAST the transcript against SwissProt (<cite>-c</cite> can be used to initiate cluster mode)<ul>
<li><a class="reference download internal" href="_downloads/runBlast.py" download=""><code class="xref download docutils literal"><span class="pre">runBlast.py</span></code></a></li>
</ul>
</li>
<li>Then, these data were are parsed and summarized.<ul>
<li><a class="reference download internal" href="_downloads/runBlastParallelParse.py" download=""><code class="xref download docutils literal"><span class="pre">runBlastParse.py</span></code></a></li>
<li><a class="reference download internal" href="_downloads/runBlastSummarize.py" download=""><code class="xref download docutils literal"><span class="pre">runBlastSummarize.py</span></code></a></li>
<li><a class="reference download internal" href="_downloads/showTaxaSummary.py" download=""><code class="xref download docutils literal"><span class="pre">showTaxaSummary.py</span></code></a></li>
</ul>
</li>
</ol>
<p>For each <em>transcript</em> the top BLAST match with the best score was kept.  The minimum BLAST e-value was set at 0.0001.</p>
<div class="section" id="swissprot-isoforms">
<h3>SwissProt (isoforms)<a class="headerlink" href="#swissprot-isoforms" title="Permalink to this headline">¶</a></h3>
<div class="align-center figure">
<a class="reference internal image-reference" href="_images/dn-trinity-blast-pie-isoforms.png"><img alt="de-novo trinity blast taxa by gene" src="_images/dn-trinity-blast-pie-isoforms.png" style="width: 630.0px; height: 489.99999999999994px;" /></a>
</div>
</div>
<div class="section" id="blast-agains-all-known-d-melanogaster-proteins">
<h3>BLAST agains all known D. melanogaster proteins<a class="headerlink" href="#blast-agains-all-known-d-melanogaster-proteins" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><div class="highlight-bash"><div class="highlight"><pre><span></span>~$ wget ftp://ftp.ensemblgenomes.org/pub/metazoa/release-28/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.28.cdna.all.fa.gz
~$ gunzip -c Drosophila_melanogaster.BDGP6.28.cdna.all.fa.gz &gt; Drome.fa
~$ makeblastdb -in Drome.fa -dbtype <span class="s1">&#39;nucl&#39;</span> -out Drome
</pre></div>
</div>
<ul class="simple">
<li><a class="reference download internal" href="_downloads/runBlastDrome.py" download=""><code class="xref download docutils literal"><span class="pre">runBlastDrome.py</span></code></a></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="blast-against-all-know-d-plexippus">
<h3>BLAST against all know D. plexippus<a class="headerlink" href="#blast-against-all-know-d-plexippus" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><blockquote>
<div><div class="highlight-bash"><div class="highlight"><pre><span></span>~$ wget ftp://ftp.ensemblgenomes.org/pub/metazoa/release-28/fasta/danaus_plexippus/cdna/Danaus_plexippus.DanPle_1.0.28.cdna.all.fa.gz
~$ gunzip -c Danaus_plexippus.DanPle_1.0.28.cdna.all.fa.gz &gt; Danaus.fa
~$ makeblastdb -in Danaus.fa -dbtype <span class="s1">&#39;nucl&#39;</span> -out Danaus
</pre></div>
</div>
</div></blockquote>
<ul class="simple">
<li><a class="reference download internal" href="_downloads/runBlastDanaus.py" download=""><code class="xref download docutils literal"><span class="pre">runBlastDanaus.py</span></code></a></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="summary">
<h3>Summary<a class="headerlink" href="#summary" title="Permalink to this headline">¶</a></h3>
<p>To summarize the BLAST results and map them to NCBI gene IDs.</p>
<blockquote>
<div><table border="1" class="docutils">
<colgroup>
<col width="32%" />
<col width="23%" />
<col width="23%" />
<col width="23%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">BLAST DB</th>
<th class="head">Transcripts</th>
<th class="head">Genes</th>
<th class="head">Proteins</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>SwissProt</td>
<td>39,457</td>
<td>11,285</td>
<td><strong>15,584</strong></td>
</tr>
<tr class="row-odd"><td><em>D. melanogaster</em></td>
<td>37,342</td>
<td>7,963</td>
<td><strong>9,672</strong></td>
</tr>
<tr class="row-even"><td><em>Danaus plexippus</em></td>
<td>7,810</td>
<td>NA</td>
<td><strong>1,392</strong></td>
</tr>
</tbody>
</table>
</div></blockquote>
</div>
<div class="section" id="links">
<h3>Links<a class="headerlink" href="#links" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><ul class="simple">
<li><a class="reference external" href="http://trinityrnaseq.sourceforge.net">Trinity Sourceforge page</a></li>
<li><a class="reference external" href="http://trinityrnaseq.sourceforge.net/analysis/full_length_transcript_analysis.html">Example from Trinity docs</a></li>
<li><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></li>
<li><a class="reference external" href="http://trinityrnaseq.sourceforge.net/#trinity_output">Trinity output</a></li>
<li><a class="reference external" href="http://trinityrnaseq.sourceforge.net/analysis/abundance_estimation.html">Trinity abundance estimation</a></li>
</ul>
</div></blockquote>
</div>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="meta-data.html" class="btn btn-neutral float-right" title="Meta-data analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="index.html" class="btn btn-neutral" title="Transcriptome supplemental materials for Pieris brassicae" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, CNRS.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>