<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Works cited &mdash; Xenopus tropicalis endurance transcriptome</title>
    
    <link rel="stylesheet" href="_static/haiku.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Xenopus tropicalis endurance transcriptome" href="index.html" />
    <link rel="prev" title="Meta-data analysis" href="results/meta-data.html" /> 
  </head>
  <body>
      <div class="header"><img class="rightlogo" src="_static/xenopus3_small.jpg" alt="Logo"/><h1 class="heading"><a href="index.html">
          <span>Xenopus tropicalis endurance transcriptome</span></a></h1>
        <h2 class="heading"><span>Works cited</span></h2>
      </div>
      <div class="topnav">
      
        <p>
        «&#160;&#160;<a href="results/meta-data.html">Meta-data analysis</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        </p>

      </div>
      <div class="content">
        
        
  <div class="section" id="works-cited">
<h1>Works cited<a class="headerlink" href="#works-cited" title="Permalink to this headline">¶</a></h1>
<table class="docutils citation" frame="void" id="anders14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Anders14]</td><td>S. Anders, P. T. Pyl &amp; W. Huber. <em>HTSeq - A Python framework to work with high-throughput sequencing data</em>,
Bioinformatics, In Print, 1-4 (2014)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="ashburner00" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Ashburner00]</td><td>M. Ashburner <em>et al</em>. <em>Gene ontology: tool for the unification of biology</em>,
Nature Genetics, 25, 25-29 (2000)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="bishop06" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Bishop06]</td><td>C. M. Bishop. <em>Pattern Recognition and Machine Learning</em>,
Springer (2006).</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="bolger14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Bolger14]</td><td>A. M Bolger, M Lohse &amp; B Usadel. <em>Trimmomatic: a flexible trimmer for Illumina sequence data</em>,
Bioinformatics,30, 2114-20 (2014)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="camacho09" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Camacho09]</td><td>C. Camacho, G. Coulouris, V. Avagyan, N  Ma, J Papadopoulos, K Bealer, &amp; T. L. Madden. <em>BLAST+: architecture and applications</em>,
BMC bioinformatics, 10, 421 (2009)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="cock09" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Cock09]</td><td>P. J. Cock <em>et al</em>. <em>Biopython: freely available Python tools for computational molecular biology and bioinformatics</em>,
Bioinformatics, 25, 1422-3 (2009)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="chang08" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Chang08]</td><td>Y-W. Chang, and C-J Lin. <em>Feature Ranking Using Linear SVM</em>,
Causation and Prediction Challenge, volume 3 of JMLR Proceedings, page 53-64. JMLR.org (2008)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="dobin13" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Dobin13]</td><td>A Dobin <em>et al</em>. <em>STAR: ultrafast universal RNA-seq aligner</em>,
Bioinformatics, 29, 15-21 (2013)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="efron07" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Efron07]</td><td>B Efron &amp; R Tibshirani. <em>On testing the significance of sets of genes</em>,
Annals of Applied Statistics, 1, 107-129 (2007)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="grabherr11" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Grabherr11]</td><td>M.G Grabherr et al. <em>Full-length transcriptome assembly from RNA-Seq data without a reference genome</em>,
Nature biotechnology, 29, 644-52 (2011)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="li09" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Li09]</td><td>H. Li <em>et al</em>. <em>The Sequence Alignment/Map format and SAMtools</em>,
Bioinformatics, 25, 2078-9 (2009)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="langmead12" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Langmead12]</td><td>B Langmead &amp; S. L. Salzberg.  <em>Fast gapped-read alignment with Bowtie 2</em>,
Nature methods, 9, 357-9 (2012)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="love14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Love14]</td><td>M. I. Love, W. Huber, &amp; S. Anders. <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</em>,
Genome biology, 15, 550 (2014)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="patro14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Patro14]</td><td>R. Patro, S. M. Mount, &amp; C Kingsford. <em>Sailfish enables alignment-free isoform quantification from RNA-seq reads using lightweight algorithms</em>,
Nature biotechnology, 32, 462-4 (2014)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="uniprotconsortium14" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[UniProtConsortium14]</td><td>U. Consortium, <em>Activities at the Universal Protein Resource (UniProt)</em>,
Nucleic acids research, 42, D191-8 (2014)</td></tr>
</tbody>
</table>
</div>


      </div>
      <div class="bottomnav">
      
        <p>
        «&#160;&#160;<a href="results/meta-data.html">Meta-data analysis</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        </p>

      </div>

    <div class="footer">
        &copy; Copyright 2015.
      Last updated on May 19, 2015.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>